domino-ui/src/main/java/org/dominokit/domino/ui/forms/Select.java

```diff
@@ -154,9 +154,20 @@ public DropDownMenu getOptionsMenu() {
   /** {@inheritDoc} */
   @Override
   public Select<T> setValue(T value, boolean silent) {
-    for (SelectOption<T> option : getOptions()) {
-      if (Objects.equals(option.getValue(), value)) {
-        select(option, silent);
+    if (isNull(value)) {
+      // same as clear & doClear but pass silent to deselect
+      if (nonNull(selectedOption)) {
```

Comment:
Instead of the nested if we can go with one if and an early exit, or even better we check for the nonNull condition and else we do the logic.